### DISCUSSION TITLE
Mechanism for proffering and voting on cloud resource archetypes

### DISCUSSION THREAD

**Discussion by AdamTylerLynch** [winglang/wing#872] [General]
Supporting multiple cloud providers bring the complexity of service selection and support. In IaC communities such as CDK and Terraform, determining where to invest engineering time on which resource types (and features) will be supported is a constant criticism and pain point.

For Wing, I would like to see a mechanism where anyone can proffer an archetype to support (ex: `bucket` or `queue`), and the community could vote. Priority would be given to those with the highest votes and a clear path for implementation. 

This could be implemented with GitHub issues and tags (Terraform AWS Provider uses this method), or another mechanism. Open to discussion.

This becomes especially important as Wing moves beyond foundational/core services and into higher level services. This also serves as a great place for contributors to go find something new to work on.

**Comment by staycoolcall911** [comment]
Great topic @AdamTylerLynch! Let me describe the mechanism we currently have. 

Anyone can open an issue on our repo and if it is a request for a new cloud resource archetype, our team will make sure to add it to the [WingSDK Roadmap](https://github.com/orgs/winglang/projects/3). 
This roadmap lists the available resource/target combinations and folks can +1 them to cast their vote. 
Later, we can view the most upvoted issues like this:
https://github.com/winglang/wing/issues?q=is%3Aissue+is%3Aopen+project%3Awinglang%2F3+sort%3Areactions-%2B1-desc

Perhaps we need to highlight that mechanism in our [Contributor's Handbook](https://docs.winglang.io/contributors/handbook).

**Comment by AdamTylerLynch** [comment]
Here is a community example that provides clear guidance, might be a good start for inspiration.
https://hashicorp.github.io/terraform-provider-aws/prioritization/
